src/sbvr-api/permissions.ts

```diff
@@ -950,18 +950,20 @@ const checkApiKey = Promise.method((req: PermissionReq, apiKey: string) => {
 	if (apiKey == null || req.apiKey != null) {
 		return;
 	}
-	return getApiKeyPermissions(apiKey)
-		.catch(err => {
-			console.warn('Error with API key:', err);
-			// Ignore errors getting the api key and just use an empty permissions object
-			return [];
-		})
-		.then(permissions => {
+	return Promise.join(
+		getApiKeyPermissions(apiKey),
+		getApiKeyActorId(apiKey),
+		(permissions: any, actor: number) => {
 			req.apiKey = {
 				key: apiKey,
 				permissions: permissions,
+				actor,
 			};
-		});
+		},
+	).catch(err => {
+		console.warn('Error with API key:', err);
+		// Ignore errors getting the api key.
```

Comment:
This now ignores errors in a different way, such that `req.apiKey` will no longer be set in the case no permissions could be fetched, I wouldn't expect it to break anything but it does have that potential

Reply:
Not sure I follow. `apiKey` will not be set either if there is an error with `getApiKeyPermissions` (as in the previous version) or with `getApiKeyActorId` (the new behavior). But every api key should be associated with an actor, to my understading.

Reply:
ok... let me guys know what I should do here :)

Reply:
I've just realised that it checks `req.apiKey != null` above and does an early return, so with this change it would mean we check the api key multiple times since we no longer set that property, so from a performance perspective it's probably worth keeping at least something we can check against

Reply:
It should be kept unchanged now (moved catch to get permissions call).